src/controllers/resources.controller.ts

```diff
@@ -1,34 +1,33 @@
-import { WithId, Document } from "mongodb";
-import { convertResultToStudyPlan, convertResultToResource } from "@utils/converter";
+import { Document, WithId } from "mongodb";
+import { convertResultToQuickLink, convertResultToResource, convertResultToStudyPlan } from "@utils/converter";
 import { getConnection } from "@utils/mongoconnection";
 import { Request, Response } from "express";
 
+const converters = new Map<string, (result: WithId<Document>) => any>([
+	["study_plans", convertResultToStudyPlan],
+	["resources", convertResultToResource],
+	["quicklinks", convertResultToQuickLink],
+]);
+
 const getAllResources = async (req: Request, res: Response) => {
 	const pageName = req.params.page;
 	if (!pageName) {
 		res.status(400).send("page name is required");
 		return;
 	}
 
-	let converterFunc: (result: WithId<Document>) => any;
-	switch (pageName) {
-		case "study_plans":
-			converterFunc = convertResultToStudyPlan;
-			break;
-		case "resources":
-			converterFunc = convertResultToResource;
-			break;
-		default:
-			res.status(400).send("invalid page name");
-			return;
-	}
+  const	converterFunc = converters.get(pageName);
+  if (!converterFunc) {
+    res.status(400).send("page not supported");
+    return;
+  }
 
 	try {
 		const client = await getConnection();
 		const data = client?.db("stamfordcenter").collection(pageName).find({});
 		const results = await data?.toArray();
 		if (!results) {
-			return res.status(404).send("file not found");
+			return res.status(404).send("content not found");
```

Comment:
Content for WHAT exactly, should specify what isn't found so we can track it down.

Reply:
changing